### PR TITLE
Update email authproof

### PR DESCRIPTION
**Pull request by mingyuan1219**
This is to update the logic to generate email authproof to fit the new contract. The two biggest changes are:
1. IdentityType and AuthType are deprecated;
2. Introduced name struct: https://github.com/HexlinkOfficial/hexlink-contracts-evm/blob/main/contracts/structs/Name.sol

Reference:
1. task to build authproof: https://github.com/HexlinkOfficial/hexlink-contracts-evm/blob/main/tasks/utils.ts#L48
3. task to calculate name hash: https://github.com/HexlinkOfficial/hexlink-contracts-evm/blob/main/test/testers.ts#L9

Note: the logic to generate twitter authproof has been deleted since it's not needed at this point.

### CODE REVIEW

File: src/web3/oracle.ts [hunk 1]
```diff
@@ -17,10 +17,9 @@ export async function genDeployAuthProof(
         throw new Error("Not connected");
     }
 
-    const authType = useAuthStore().user!.authType;
     const identityType = useAuthStore().user!.identityType;
     let genAuthProof: HttpsCallable;
-    if (identityType === "twitter.com" && authType === "oauth") {
+    if (identityType === "twitter.com") {
         genAuthProof = httpsCallable(functions, 'genTwitterOAuthProof');
```
Comment (dongshu2013) [inline src/web3/oracle.ts:23]: should we throw here since we removed the genTwitterOAuthProof?

File: functions/src/verifier.ts [hunk 2]
```diff
@@ -77,24 +55,49 @@ const genAuthProof = async (uid: string, identity: string,
...
     requestId: requestId,
-    issuedAt: issuedAt,
-    identityType: identity,
-    authType: auth,
-    signature: encodedSig,
+    expiredAt: expiredAt,
+    validator: validator,
+    proof: encodedSig,
   };
   return {code: 200, authProof: AuthProof};
```
Comment (dongshu2013) [inline functions/src/verifier.ts:70]: maybe we can simply return the encodedSig here

**Review by dongshu2013 — APPROVED** [review]
We will need to update the deployment logic but we can do it in another diff

File: functions/src/verifier.ts (hunk 2 above)
Comment (mingyuan1219) [reply functions/src/verifier.ts]: makes sense.

File: src/web3/oracle.ts (hunk 1 above)
Comment (mingyuan1219) [reply src/web3/oracle.ts:23]: Good point. Will update it in another diff along with the update of deployment logic.

**Pull request merged**